### DISCUSSION TITLE
Events aren't printing

### DISCUSSION THREAD

**Discussion by Yonygreen** [kuzmoyev/google-calendar-simple-api#165] [General]
I just tried using this library for the first time. I went through all the steps, but when trying to view events like in the example, nothing is printed at all. this is my code:
```
from gcsa.google_calendar import GoogleCalendar

gc = GoogleCalendar()
for event in gc:
    print(event)
```

When trying to use the same gc object to create an event like in the event creation example, it does work and it creates the event perfectly.
Thanks in advanced!

**Comment by kuzmoyev** [comment]
A bit obvious question, but do you have events in the calendar? If yes, on what dates?

Could you please also post the example of the event you've created and the code you've used?

Note that iterating over the calendar this way only prints the events for the following year (i.e. from today to the same date next year)

**Reply by Yonygreen** [reply]
It's a workplace calendar, I have multiple events on it each day (and a lot of recurring events).

```
gc = GoogleCalendar()
event = Event(
    "Test",
    start=datetime(2023,3,6,16,0)

)

gc.add_event(event)

```
I am attaching a photo of part of today's calendar, the event created with the script is "Test".
![Inkedchrome_ipHIqSv9Ak](https://user-images.githubusercontent.com/49966039/223077829-dbdcd939-4dc3-4d1b-90f5-dba06fbc8ced.jpg)

**Comment by kuzmoyev** [comment]
So `print(list(gc))` would produce an empty list (`[]`)? Or do you get some errors?

**Reply by Yonygreen** [reply]
Yes, I get an empty list `[]`

**Reply by kuzmoyev** [reply]
Can you try a bigger time range, say
```
for e in gc[datetime(2000,3,6,16,0): datetime(2025,3,6,16,0)]:
    print(e)
```

**Reply by Yonygreen** [reply]
When running this code I do get some events. I figured out that it only prints events from a specific calendar which I don't use anymore.
When running this code: 
```
gc = GoogleCalendar()
for cal in gc.get_calendar_list():
    print(cal)
```
I get this output:
```
None - (Groove Music Center)
None - (C)
None - (D)
None - (E)
None - (Recording studio)
None - (Did not arrive)
None - (Shifts)
None - (A)
None - (F)
None - (B)
```
The calendar I get events from when running your suggested code is the first one, Groove Music Center, but the calendars I actually need to get event data for are A,B,C,D,E,F.
I can get each calendar with:
`gc.get_calendar(cal_id)`
but I can't interact with events with this calendar item as far as I understand.
Thanks in advanced for all the previous and further help!

**Reply by kuzmoyev** [reply]
Right. The `GoogleCalendar` by default uses your "primary" calendar. You have two options to use different calendars:

1. Specify `default_calendar` while initializing:
```python
gc = GoogleCalendar(default_calendar='<calendar_id>')
```

or probably a better-suited option for your case:

2. Specify `calendar_id` in `get_events`:

```python
CALENDARS = ('A', 'B', 'C', 'D', 'E', 'F')

gc = GoogleCalendar()

for cal in gc.get_calendar_list():
    if cal.summary in CALENDARS:
        print(f'Events in {cal.summary}:')
        for e in gc.get_events(calendar_id=cal.id):
            print(e)
```
